src/Microsoft.Data.Analysis/DataFrame.cs

```diff
@@ -485,7 +487,30 @@ public DataFrame Append(IEnumerable<object> row = null, bool inPlace = false)
                     }
                     if (value != null)
                     {
-                        value = Convert.ChangeType(value, column.DataType);
+                        try
+                        {
+                            value = Convert.ChangeType(value, column.DataType, cultureInfo);
+                        }
+                        catch(FormatException)
+                        {
+                            Console.Write($"Value \"{value}\" cannot be converted to type {column.DataType} (Column name: {column.Name}). ");
```

Comment:
Related to #2902 , give a message to the user what value can't be converted to which type. If Double/Single, assign a `NaN` value instead. Otherwise throw a `FormatException` as previously ( this also doesn't break unit test `TestAppendRow` which checks for `FormatException` ).